lib/rules/valid-v-for.js

```diff
@@ -19,34 +19,50 @@ const utils = require('../utils')
  * Check whether the given attribute is using the variables which are defined by `v-for` directives.
  * @param {ASTNode} vFor The attribute node of `v-for` to check.
  * @param {ASTNode} vBindKey The attribute node of `v-bind:key` to check.
+ * @param {boolean} isChild If checks on child in template[v-for].
  * @returns {boolean} `true` if the node is using the variables which are defined by `v-for` directives.
  */
-function isUsingIterationVar (vFor, vBindKey) {
+function isUsingIterationVar (vFor, vBindKey, isChild) {
   if (vBindKey.value == null) {
     return false
   }
   const references = vBindKey.value.references
   const variables = vFor.parent.parent.variables
-
-  return references.some(reference =>
+  const used = references.some(reference =>
         variables.some(variable =>
             variable.id.name === reference.id.name &&
             variable.kind === 'v-for'
         )
     )
+
+  if (!isChild || used) {
+    return used
+  }
+  const childFor = utils.getDirective(vBindKey.parent.parent, 'for')
+  if (!childFor) {
+    return false
+  }
+  const childForRefs = childFor.value.references
+  return childForRefs.some(cref =>
+         variables.some(variable =>
+             cref.id.name === variable.id.name &&
+             variable.kind === 'v-for'
+         )
+    )
 }
 
 /**
  * Check the given element about `v-bind:key` attributes.
  * @param {RuleContext} context The rule context to report.
  * @param {ASTNode} vFor The attribute node of `v-for` to check.
  * @param {ASTNode} element The element node to check.
+ * @param {boolean} isChild If element is a child of template[v-for].
  */
-function checkKey (context, vFor, element) {
+function checkKey (context, vFor, element, isChild) {
```

Comment:
checkKey is not always called with 4 parameters

Reply:
labeled as optional